packages/codemods/src/v5/is-loading/is-loading.js

```diff
@@ -0,0 +1,201 @@
+   */
+  const findRestElementInObjectPattern = (node) => {
+    return (
+      node.properties.find((property) =>
+        jscodeshift.match(property, {
+          type: jscodeshift.RestElement.name,
+        }),
+      ) ?? null
+    )
+  }
+
+  const replacer = (path, transformNode) => {
+    const node = path.node
+    const parentNode = path.parentPath.value
+    const { start, end } = getNodeLocation(node)
+
+    try {
+      if (!isVariableDeclarator(parentNode)) {
+        // The parent node is not a variable declarator, the transformation will be skipped.
+        return node
+      }
+
+      const lookupNode = path.scope.node
+      const variableDeclaratorId = parentNode.id
+
+      if (isObjectPattern(variableDeclaratorId)) {
+        const isLoadingObjectProperty =
+          findIsLoadingObjectPropertyInObjectPattern(variableDeclaratorId)
+
+        if (isLoadingObjectProperty) {
+          jscodeshift(lookupNode)
+            .find(jscodeshift.Identifier, {
+              name: originalName,
+            })
+            .replaceWith(({ node: mutableNode }) => {
+              mutableNode.name = newName
+
+              return mutableNode
+            })
+        }
+
+        const restElement = findRestElementInObjectPattern(variableDeclaratorId)
+
+        if (restElement) {
+          findIsLoadingPropertiesOfIdentifier(
+            lookupNode,
+            restElement.argument,
+          ).replaceWith(({ node: mutableNode }) => {
+            mutableNode.property.name = newName
+
+            return mutableNode
+          })
+        }
+
+        return node
+      }
+
+      if (utils.isIdentifier(variableDeclaratorId)) {
+        findIsLoadingPropertiesOfIdentifier(
+          lookupNode,
+          variableDeclaratorId,
+        ).replaceWith(({ node: mutableNode }) => {
+          mutableNode.property.name = newName
+
+          return mutableNode
+        })
+
+        return node
+      }
+
+      utils.warn(
+        `The usage in file "${filePath}" at line ${start}:${end} could not be transformed. Please migrate this usage manually.`,
+      )
+
+      return node
+    } catch (error) {
+      utils.warn(
+        `An unknown error occurred while processing the "${filePath}" file. Please review this file, because the codemod couldn't be applied.`,
+      )
+
+      return node
+    }
+  }
+
+  createUseQueryLikeTransformer({ jscodeshift, utils, root }).execute(
+    config.hooks,
+    replacer,
+  )
+
+  createQueryClientTransformer({ jscodeshift, utils, root }).execute(
+    config.queryClientMethods,
+    replacer,
+  )
+}
+
+module.exports = (file, api) => {
+  const jscodeshift = api.jscodeshift
+  const root = jscodeshift(file.source)
+  const utils = createUtilsObject({ root, jscodeshift })
+  const filePath = file.path
+
+  const dependencies = { jscodeshift, utils, root, filePath }
+
+  transformUsages({
+    ...dependencies,
+    config: {
+      hooks: ['useQuery'],
```

Comment:
@TkDodo what other hooks are affected?

Reply:
`useMutation` is also affected

Reply:
https://github.com/TanStack/query/pull/6388/commits/66035671e34e06c064e2acfdcc3fb4136988f933